test/unitTest/java/org/thoughtcrime/securesms/NicknameMocks.java

```diff
@@ -0,0 +1,61 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+
+import org.junit.runner.RunWith;
+import org.mockito.BDDMockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.RecipientDatabase;
+import org.thoughtcrime.securesms.recipients.Recipient;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockingDetails;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DatabaseFactory.class})
+public class NicknameMocks extends BaseUnitTest {
+    protected DatabaseFactory   databaseFactory;
+    protected Recipient         recipient;
+    protected Recipient         recipient2;
+    protected RecipientDatabase recipientDatabase;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        databaseFactory = mock(DatabaseFactory.class);
+        recipient = mock(Recipient.class);
+        recipient2 = mock(Recipient.class);
+        recipientDatabase = mock(RecipientDatabase.class);
+    }
+
+    protected void setupSetNicknameMethod() {
+        when(recipientDatabase.setNickname(recipient, "test")).thenReturn(true)
+                .thenReturn(false);
+
+        when(recipientDatabase.setNickname(recipient, "new test")).thenReturn(true)
+                .thenReturn(false);
+
+    }
+
+    // TODO add removeNickname mock
```

Comment:
should do this before merge

Reply:
no, we need to merge, so that @Dan-Yoo starts working on his "remove" feature and write his own tests

Reply:
So remove it from here ?

Reply:
we will need to talk to @Dan-Yoo first,

Reply:
if we remove it @Dan-Yoo will spend time trying to understand and re designing the wheel. in this case, he will right away go to the solution.

Reply:
... Or we can communicate and he will know what to do ?